### PR TITLE
Mandatory checkpoint on Canopy - breaking cached state test change

### LINKED ISSUE #1846: Change the mandatory checkpoint to Canopy
**Is your feature request related to a problem? Please describe.**

Zebra should change its mandatory checkpoint to the `Canopy` network upgrade, to reduce the work we need to do for `NU5` activation.

Checkpointing on `Canopy` lets us skip the following validation:
* transfers into the Sprout value pool (rejected at `Canopy` activation)
* Founders Reward (finished right before `Canopy` activation)
* validation of the `PreSapling`, `Sapling`, and `ChainHistoryActivation` (Heartwood activation) `RootHash` variants
* the pre-`Blossom` block target spacing (finished right before `Blossom` activation)

**Describe the solution you'd like**

- [ ] Update the list of Zebra checkpoints for mainnet and testnet
- [ ] Change the `Sapling` mandatory checkpoint to `Canopy`

**Describe alternatives you've considered**

Write a whole lot of validation code that we'll never really use.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

We want to checkpoint in Canopy to avoid the development of some validation code. More details in https://github.com/ZcashFoundation/zebra/issues/1846 

## Solution

Change the mandatory checkpoint from Sapling to Canopy in the testnet and in the mainnet.

I synchronized zebra using this PR up to tip in 2 hours and 40 minutes. This is an improve (expected) from the last tries in the same machine of 3 hours.

In this PR mainnet checkpoints were upgraded while testnet ones were already passing Canopy so this was not needed.

## Review

@teor2345 

## Related Issues

If merged this PR will close https://github.com/ZcashFoundation/zebra/issues/1846

## Follow Up Tasks

Post-canopy test vectors #1104 and #1099
Panic on pre-Canopy blocks in `NonFinalizedState::commit_block` #1903

### CODE REVIEW

**Comment by teor2345** [comment]
> In this PR mainnet checkpoints were upgraded while testnet ones were already passing Canopy so this was not needed.

Please update both mainnet and testnet checkpoints for consistency. Otherwise, our testnet checkpoints will be very old.

File: zebra-consensus/src/checkpoint/list/tests.rs [hunk 1]
```diff
@@ -241,28 +241,28 @@ fn checkpoint_list_load_hard_coded() -> Result<(), BoxError> {
...
 }
 
 #[test]
-fn checkpoint_list_hard_coded_sapling_testnet() -> Result<(), BoxError> {
-    checkpoint_list_hard_coded_sapling(Testnet)
+fn checkpoint_list_hard_coded_canopy_testnet() -> Result<(), BoxError> {
+    checkpoint_list_hard_coded_canopy(Testnet)
 }
 
 /// Check that the hard-coded lists cover the Sapling network upgrade
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/list/tests.rs]:
```suggestion
/// Check that the hard-coded lists cover the Canopy network upgrade
```

File: zebra-consensus/src/checkpoint/list/tests.rs [hunk 2]
```diff
@@ -241,28 +241,28 @@ fn checkpoint_list_load_hard_coded() -> Result<(), BoxError> {
...
-fn checkpoint_list_hard_coded_sapling(network: Network) -> Result<(), BoxError> {
+fn checkpoint_list_hard_coded_canopy(network: Network) -> Result<(), BoxError> {
     zebra_test::init();
 
-    let sapling_activation = Sapling
+    let sapling_activation = Canopy
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/list/tests.rs]:
```suggestion
    let canopy_activation = Canopy
```

**Review by teor2345 — CHANGES_REQUESTED** [review]
Looks good, just missing an update to the testnet checkpoints. (And a few typos.)

I already have that synced, so I'll start generating the list now, and push it to this branch when it's done.

I also want to checkout this PR, and make sure we've replaced every relevant instance of Sapling with Canopy.

**Comment by teor2345** [comment]
> > In this PR mainnet checkpoints were upgraded while testnet ones were already passing Canopy so this was not needed.
> 
> Please update both mainnet and testnet checkpoints for consistency. Otherwise, our testnet checkpoints will be very old.

I pushed this change.

In general, we want to keep both checkpoint lists up to date, whenever we change one of them.

**Comment by teor2345** [comment]
> I also want to checkout this PR, and make sure we've replaced every relevant instance of Sapling with Canopy.

There were lots of missing changes.

Please use an interactive search and replace tool to create future PRs like this.

I used the following commands:
```sh
fastmod Sapling Canopy
fastmod sapling canopy
fastmod sapling canopy .github
```

Then I checked each change to make sure that it was correct. (Sapling is the name of a network upgrade, and a shielded pool, so we can't just do an automatic search and replace.)

The `.github` command was required because `fastmod` ignores dotfiles by default.
(I also needed to remove smart quotes, due to a bug in `fastmod`'s unicode handling. See #1902.)

**Comment by teor2345** [comment]
@dconnolly this PR changes the cached tests to use Canopy, so it will require a cached state rebuild.

Can you show @oxarbitrage how to do that?

File: book/src/dev/rfcs/0005-state-updates.md [hunk 3]
```diff
@@ -419,7 +419,7 @@ chain and updates all side chains to match.
 
 Commit `block` to the non-finalized state.
 
-1. If the block is a pre-Sapling block, panic.
+1. If the block is a pre-Canopy block, panic.
```
Comment (teor2345) [inline book/src/dev/rfcs/0005-state-updates.md:422]: I can't actually see where we do this check in the code, maybe we need to open a follow-up issue.

**Review by teor2345 — COMMENTED** [review]
@oxarbitrage can you check my changes to make sure they're ok?

I also added some follow-up tasks.

File: book/src/dev/rfcs/0005-state-updates.md (hunk 3 above)
Comment (teor2345) [reply book/src/dev/rfcs/0005-state-updates.md:422]: This is now #1903

**Review by oxarbitrage — COMMENTED** [review]
- I didn't changed the RFCs in the initial PR because i wasn't sure if that was something we should do, for example i am unsure if a ZIP can be changed after is implemented. In this case i think is ok.
- I didn't added new testnet checkpoints by lack of time on Friday(had to sync a zcashd testnet), thanks for adding them.
- I missed some changes, thanks for the fixes.

Looks good.

**Review by teor2345 — APPROVED** [review]
@oxarbitrage has checked my changes, and I've checked his.

Leaving this PR open so @dconnolly can check the cached state changes if she wants.

**Comment by teor2345** [comment]
>     * I didn't changed the RFCs in the initial PR because i wasn't sure if that was something we should do, for example i am unsure if a ZIP can be changed after is implemented. In this case i think is ok.

In general, I don't think we need to spend a lot of effort updating RFCs - if there is a significant change, we should do a rewrite or a new RFC.

But in this case, it took only a few minutes to update. And I didn't want us to get confused between Sapling and Canopy checkpoints.

**Review by teor2345 — APPROVED** [review]
This is fine

**Pull request merged**